dashboard/src/shared/Kube.ts

```diff
@@ -0,0 +1,40 @@
+import { axios } from "./Auth";
+import { IResource, ISecret } from "./types";
+
+export const KUBE_ROOT_URL = "/api/kube";
+
+export class Kube {
+  public static getResourceURL(
+    apiVersion: string,
+    resource: string,
+    namespace?: string,
+    name?: string,
+    query?: string,
+  ) {
+    let url = `${KUBE_ROOT_URL}/${apiVersion === "v1" ? "api/v1" : `apis/${apiVersion}`}`;
+    if (namespace) {
+      url += `/namespaces/${namespace}`;
+    }
+    url += `/${resource}`;
+    if (name) {
+      url += `/${name}`;
+    }
+    if (query) {
+      url += `?${query}`;
+    }
+    return url;
+  }
+
+  public static async getResource(
+    apiVersion: string,
+    resource: string,
+    namespace?: string,
+    name?: string,
+    query?: string,
+  ) {
+    const { data } = await axios.get<IResource | ISecret>(
```

Comment:
hmm, are we going to have to enumerate all the types here? Is there a better way we can keep the type of the resource?